--- texinfo-6.3/info/infodoc.c.0107~	2016-02-06 13:20:23.000000000 +0100
+++ texinfo-6.3/info/infodoc.c	2016-09-13 00:34:27.561440799 +0200
@@ -80,6 +80,10 @@ static char *info_internal_help_text[] =
   NULL
 };
 
+static char *where_is_rep = NULL;
+static int where_is_rep_index = 0;
+static int where_is_rep_size = 0;
+
 static char *where_is_internal (Keymap map, InfoCommand *cmd);
 
 static void
@@ -214,6 +218,8 @@ create_internal_info_help_node (int help
     {
       InfoCommand *cmd = &function_doc_array[i];
 
+      where_is_rep_index = 0;
+
       if (cmd->func != info_do_lowercase_version
           && !where_is_internal (info_keymap, cmd)
           && !where_is_internal (echo_area_keymap, cmd))
@@ -818,9 +824,6 @@ replace_in_documentation (const char *st
 
 /* Return a string of characters which could be typed from the keymap
    MAP to invoke FUNCTION. */
-static char *where_is_rep = NULL;
-static int where_is_rep_index = 0;
-static int where_is_rep_size = 0;
 
 char *
 where_is (Keymap map, InfoCommand *cmd)
